test/lib.include

```diff
@@ -74,9 +74,38 @@ fsed() {
 }
 
 restart_docker() {
-	docker-compose stop node
-	docker-compose start node
-	sleep 10
+	# Behold!
+	#
+	# You thought stop/start restarts? How wrong. Restart restarts.
+	docker compose restart
+	# The suspense!
+	#
+	# Wait until compose believes the health probe succeeds.
+	#
+	# The container may decide to recompile, or grab a coffee at crates.io, or
+	# whatever. In any case, restart doesn't mean the server is up yet.
+	docker compose up --no-recreate --detach --wait-timeout 60
+	# Belts and suspenders!
+	#
+	# The health probe runs inside the container, but that doesn't mean we can
+	# reach it from outside. Ping until we get through.
+	ping
```

Comment:
Am I correct in asserting that the minimal set of changes needed to remove the flakiness from those tests were
1. Not running them in parallel
2. This ping utility

Reply:
Also `restart`. `stop` followed by `start` works only some of the time.